README.md

```diff
@@ -4,132 +4,949 @@
+- **Tool Execution**: Executes tools provided by MCP servers or custom implementations
+- **Session Management**: Maintains conversation state and context across multiple interactions
+- **Streaming Support**: Provides real-time streaming of agent responses
+- **Memory Management**: Handles short-term and long-term memory for agents
+
+**Supported Frameworks:**
+- **ADK (Agent Development Kit)**: Google's official agent framework (primary)
+- **CrewAI**: Multi-agent orchestration framework
+- **LangGraph**: State machine-based agent workflows
+
+**Technical Stack:**
+- Implemented in Python for optimal ML/AI library support
+- Containerized for consistent deployment across environments
+- Supports both declarative and BYO (Bring-Your-Own) agent types
+
+### 3. Web UI
+
+The **Web UI** is a modern, responsive web application built with Next.js that provides a comprehensive interface for managing and interacting with kagent.
+
+**Features:**
+- **Agent Management**: Create, configure, and manage agents through an intuitive interface
+- **Conversation Interface**: Chat with agents in real-time with streaming support
+- **Session History**: View and search through past conversations
+- **Tool Configuration**: Configure and manage MCP servers and tools
+- **Model Configuration**: Set up and manage LLM provider configurations
+- **Observability Dashboards**: Visualize metrics, traces, and logs
+- **Multi-tenant Support**: Support for multiple namespaces and teams
 
-- **Controller**: The controller is a Kubernetes controller that watches the kagent custom resources and creates the necessary resources to run the agents.
-- **UI**: The UI is a web UI that allows you to manage the agents and tools.
-- **Engine**: The engine runs your agents using [ADK](https://google.github.io/adk-docs/).
-- **CLI**: The CLI is a command-line tool that allows you to manage the agents and tools.
+**Technical Stack:**
+- **Frontend**: Next.js 14+ with React and TypeScript
+- **Styling**: Tailwind CSS with shadcn/ui components
+- **State Management**: React hooks and context API
+- **API Integration**: RESTful API communication with the Controller
 
-## Get Involved
+### 4. CLI
 
-_We welcome contributions! Contributors are expected to [respect the kagent Code of Conduct](https://github.com/kagent-dev/community/blob/main/CODE-OF-CONDUCT.md)_
+The **CLI** (`kagent`) is a comprehensive command-line tool that provides full control over kagent operations from the terminal.
+
+**Key Commands:**
+- `kagent agent deploy`: Deploy agents from YAML manifests or templates
+- `kagent agent list`: List all agents in a namespace
+- `kagent agent invoke`: Invoke agents directly from the command line
+- `kagent mcp`: Manage MCP server configurations
+- `kagent modelconfig`: Configure LLM providers
+- `kagent install`: Install kagent in a Kubernetes cluster
+
+**Installation:**
+```bash
+curl https://raw.githubusercontent.com/kagent-dev/kagent/refs/heads/main/scripts/get-kagent | bash
```

Comment:
this should be and I think we have it in the quick start at the start of the doc